ext/re2/re2.cc

```diff
@@ -1418,8 +1410,8 @@ static VALUE re2_Replace(VALUE self, VALUE str, VALUE pattern,
     RE2::Replace(&str_as_string, StringValuePtr(pattern),
         StringValuePtr(rewrite));
 
-    return ENCODED_STR_NEW2(str_as_string.data(), str_as_string.size(),
-        pattern);
+    return ENCODED_STR_NEW(str_as_string.data(), str_as_string.size(),
```

Comment:
Given this change, we should document how RE2 only supports UTF-8 and ISO-8859-1 and anything else is undefined.